xdsl/dialects/llvm.py

```diff
@@ -107,30 +107,112 @@ def is_typed(self):
 
 @irdl_op_definition
 class GEPOp(Operation):
+    """
+    The getelementptr is rather difficult to understand. There's a lot of
+    good literature online on how this works exactly. Here's my attempt
+    at explaining how it works and doesn't work using a few examples:
+
+    But first, you need to always keep in mind that:
+    > GEP never dereferences, it only does math on the given pointer
+
+    It *always* returns a pointer to the element you "selected"!
```

Comment:
This part probably needs rewriting for a number of reasons.
Please avoid, me/you/my etc.  This part is not really giving any useful info I think